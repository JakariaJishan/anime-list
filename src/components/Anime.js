import React from "react";
import { NavLink } from "react-router-dom";

function Anime({ anime }) {
  const { mal_id, title, episodes, images } = anime;

  return (
    <NavLink to={`/episodes/${mal_id}`}>
      <img src={images.jpg.image_url}></img>
      {title}
      {episodes}

      
    </NavLink>
  );
}

export default Anime;
