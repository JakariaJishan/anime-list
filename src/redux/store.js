import { configureStore } from '@reduxjs/toolkit';
import animeSlice from './Anime-list/animeSlice';
import episodeSlice from './Anime-list/episodeSlice';

export default configureStore({
  reducer: {
    animeList: animeSlice,
    episodeList: episodeSlice,
  },
});
