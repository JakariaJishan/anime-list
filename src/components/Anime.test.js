import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "../redux/store";
import Anime from "./Anime";

const anime = {
  mal_id: 1,
  title: "Cowboy Bebop",
  episodes: 26,
  images: {
    jpg: { image_url: "https://cdn.myanimelist.net/images/anime/4/19644.jpg" },
  },
};

describe("test suite for Anime component", () => {
  it("expects to render a snapshot of the Anime component", () => {
    const animeTest = render(
      <Provider store={store}>
        <BrowserRouter>
          <Anime anime={anime} />
        </BrowserRouter>
      </Provider>
    );
    expect(animeTest).toMatchSnapshot();
  });
});
