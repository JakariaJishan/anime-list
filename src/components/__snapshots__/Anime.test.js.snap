// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for Anime component expects to render a snapshot of the Anime component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class=""
        href="/episodes/1"
      >
        <div
          class="rightArrow"
        >
          <img
            alt="right arrow icon"
            src="right-arrow.png"
          />
        </div>
        <div
          class="card"
        >
          <img
            height="200px"
            src="https://cdn.myanimelist.net/images/anime/4/19644.jpg"
            width="150px"
          />
          <div
            class="cardTitle"
          >
            <h3>
              Cowboy Bebop
            </h3>
            <h5>
              Total Episodes : 
              26
            </h5>
          </div>
        </div>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class=""
      href="/episodes/1"
    >
      <div
        class="rightArrow"
      >
        <img
          alt="right arrow icon"
          src="right-arrow.png"
        />
      </div>
      <div
        class="card"
      >
        <img
          height="200px"
          src="https://cdn.myanimelist.net/images/anime/4/19644.jpg"
          width="150px"
        />
        <div
          class="cardTitle"
        >
          <h3>
            Cowboy Bebop
          </h3>
          <h5>
            Total Episodes : 
            26
          </h5>
        </div>
      </div>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
